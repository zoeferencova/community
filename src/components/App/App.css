.App {
    text-align: center;
    height: 100%;
    width: 100%;
    position: absolute;
    bottom: 0;
    top: 0;
    left: 0;
    right: 0;
    display: flex;
    flex-direction: column;
    margin: 0;
    padding: 0;
    overflow-x: hidden;
}