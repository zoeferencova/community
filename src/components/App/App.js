import React, { Component } from "react";
import TokenService from "../../services/token-service";

import UnauthenticatedApp from "../UnauthenticatedApp/UnauthenticatedApp";
import AuthenticatedApp from "../AuthenticatedApp/AuthenticatedApp";
import Nav from "../Nav/Nav";
// import TokenService from "../../services/token-service";

import "./App.css"
import { withRouter } from "react-router-dom";

class App extends Component {
  state = {
    isLoggedIn: TokenService.hasAuthToken() ? true : false,
  }

  setLoggedIn = (status) => {
    this.setState({ isLoggedIn: status })
  }

  render() {
    return ( 
        <main className="App">
          <Nav isLoggedIn={this.state.isLoggedIn} />
          {this.state.isLoggedIn ? <AuthenticatedApp history={this.props.history} setLoggedIn={this.setLoggedIn} /> : <UnauthenticatedApp history={this.props.history} setLoggedIn={this.setLoggedIn} />}
        </main>
    );
  }
}

export default withRouter(App);