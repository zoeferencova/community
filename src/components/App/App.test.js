import React from "react";
import { mount } from "enzyme";
import App from "./App";
import { BrowserRouter } from "react-router-dom";

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });
  
  it("renders App component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders only the UnauthenticatedApp component if the isLoggedIn state is false", () => {
    wrapper.find("App").setState({ isLoggedIn: false })
    expect(wrapper.find("UnauthenticatedApp").length).toBe(1);
    expect(wrapper.find("AuthenticatedApp").length).toBe(0);
  });

  it("renders only the AuthenticatedApp component if the isLoggedIn state is true", () => {
    wrapper.find("App").setState({ isLoggedIn: true })
    expect(wrapper.find("AuthenticatedApp").length).toBe(1);
    expect(wrapper.find("UnauthenticatedApp").length).toBe(0);
  });

  it("passing false to setLoggedIn updates the isLoggedIn state to false", () => {
    wrapper.find("App").instance().setLoggedIn(false);
    expect(wrapper.find("App").state().isLoggedIn).toEqual(false);
  });

  it("passing true to setLoggedIn updates the isLoggedIn state to true", () => {
    wrapper.find("App").instance().setLoggedIn(true);
    expect(wrapper.find("App").state().isLoggedIn).toEqual(true);
  });
})



