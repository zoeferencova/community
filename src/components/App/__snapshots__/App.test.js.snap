// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(App)>
      <App
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <main
          className="App"
        >
          <UnauthenticatedApp
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isLoggedIn={false}
            setLoggedIn={[Function]}
          >
            <main
              className="main"
            >
              <Nav
                isLoggedIn={false}
              >
                <nav
                  className="nav"
                >
                  <NavLink
                    id="loggedOut"
                    to="/"
                  >
                    <Link
                      aria-current="page"
                      className="active"
                      id="loggedOut"
                      style={Object {}}
                      to={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": null,
                        }
                      }
                    >
                      <LinkAnchor
                        aria-current="page"
                        className="active"
                        href="/"
                        id="loggedOut"
                        navigate={[Function]}
                        style={Object {}}
                      >
                        <a
                          aria-current="page"
                          className="active"
                          href="/"
                          id="loggedOut"
                          onClick={[Function]}
                          style={Object {}}
                        >
                          <img
                            alt="logo"
                            className="bigLogo"
                            src="big-logo.png"
                          />
                        </a>
                      </LinkAnchor>
                    </Link>
                  </NavLink>
                  <ul
                    className="navLinks"
                    id="loggedOut"
                  >
                    <NavLink
                      to="/login"
                    >
                      <Link
                        aria-current={null}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/login",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          href="/login"
                          navigate={[Function]}
                        >
                          <a
                            aria-current={null}
                            href="/login"
                            onClick={[Function]}
                          >
                            <li
                              className="logIn"
                            >
                              Log in
                            </li>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                    <NavLink
                      to="/register"
                    >
                      <Link
                        aria-current={null}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/register",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          href="/register"
                          navigate={[Function]}
                        >
                          <a
                            aria-current={null}
                            href="/register"
                            onClick={[Function]}
                          >
                            <li
                              className="signUp"
                            >
                              Sign up
                            </li>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                  </ul>
                </nav>
              </Nav>
              <withRouter(ErrorBoundary)
                key="/"
              >
                <ErrorBoundary
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <Switch>
                    <Route
                      component={[Function]}
                      computedMatch={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      exact={true}
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      path="/"
                    >
                      <component
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        match={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                      >
                        <withRouter(LandingPage)
                          isLoggedIn={false}
                          setLoggedIn={[Function]}
                        >
                          <LandingPage
                            history={
                              Object {
                                "action": "POP",
                                "block": [Function],
                                "createHref": [Function],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                            }
                            isLoggedIn={false}
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                            match={
                              Object {
                                "isExact": true,
                                "params": Object {},
                                "path": "/",
                                "url": "/",
                              }
                            }
                            setLoggedIn={[Function]}
                          >
                            <main
                              className="main"
                            >
                              <header
                                className="header"
                              >
                                <div
                                  className="left"
                                >
                                  <h1
                                    className="heading"
                                  >
                                    Coming together in a time of need
                                  </h1>
                                  <p
                                    className="subheading"
                                  >
                                    Right now, more than ever, we need to unite our communities and help those in need. CommUnity is a platform that enables the sharing of services and resources to ensure no one is left stranded during the Coronavirus (COVID-19) outbreak.
                                  </p>
                                  <div
                                    className="buttonSection"
                                  >
                                    <button
                                      className="signupButton"
                                      onClick={[Function]}
                                      type="button"
                                    >
                                      Start now
                                    </button>
                                    <button
                                      className="demoButton"
                                      onClick={[Function]}
                                      type="button"
                                    >
                                      See a demo
                                    </button>
                                  </div>
                                </div>
                                <div
                                  className="right"
                                >
                                  <img
                                    alt="grocery delivery"
                                    className="banner desktop"
                                    src="banner.jpeg"
                                  />
                                </div>
                              </header>
                              <section
                                className="section first"
                              >
                                <div
                                  className="left"
                                >
                                  <h2
                                    className="heading"
                                  >
                                    How does it work?
                                  </h2>
                                  <p
                                    className="subheading"
                                  >
                                    Post offers or requests for help and see other active postings in your neighborhood. 
                                    <br />
                                    <br />
                                     Other members of your community can then offer to help or accept your offer, initiating a private chat where you can securely work out the details.
                                  </p>
                                </div>
                                <div
                                  className="right"
                                >
                                  <img
                                    alt="mockup of application on desktop browser"
                                    className="mockupDesktop"
                                    src="mockup1.png"
                                  />
                                </div>
                              </section>
                              <section
                                className="section icons"
                              >
                                <h2>
                                  What our helpers can do for you
                                </h2>
                                <div
                                  className="iconList"
                                >
                                  <div
                                    className="iconSection"
                                  >
                                    <img
                                      alt="man handing supplies to older couple"
                                      className="iconImage"
                                      src="supplies.jpg"
                                    />
                                    <div
                                      className="iconColumn"
                                    >
                                      <h3
                                        className="iconHeading"
                                      >
                                        Pick up supplies
                                      </h3>
                                      <p
                                        className="iconSubheading"
                                      >
                                        Running low on groceries, medicine, or household supplies? Our helpers can grab anything you need and drop it off at your doorstep.
                                      </p>
                                    </div>
                                  </div>
                                  <div
                                    className="iconSection"
                                  >
                                    <img
                                      alt="woman walking dog wearing a face mask"
                                      className="iconImage"
                                      src="dog.jpg"
                                    />
                                    <div
                                      className="iconColumn"
                                    >
                                      <h3
                                        className="iconHeading"
                                      >
                                        Walk your dog
                                      </h3>
                                      <p
                                        className="iconSubheading"
                                      >
                                        Our helpers can take your furry friend for a safe stroll if you don’t feel comfortable being out and about.
                                      </p>
                                    </div>
                                  </div>
                                  <div
                                    className="iconSection"
                                  >
                                    <img
                                      alt="decorative shape"
                                      className="shape1"
                                      src="shape1.png"
                                    />
                                    <img
                                      alt="decorative shape"
                                      className="shape2"
                                      src="shape2.png"
                                    />
                                    <img
                                      alt="woman talking on the phone looking at laptop"
                                      className="iconImage"
                                      src="phone.jpg"
                                    />
                                    <div
                                      className="iconColumn"
                                    >
                                      <h3
                                        className="iconHeading"
                                      >
                                        Friendly chat
                                      </h3>
                                      <p
                                        className="iconSubheading"
                                      >
                                        We know that quarantining can be an isolating and frightening time. Our helpers can offer a friendly chat through our secure instant messaging system.
                                      </p>
                                    </div>
                                  </div>
                                </div>
                              </section>
                              <section
                                className="section security"
                              >
                                <div
                                  className="left"
                                >
                                  <img
                                    alt="mockup of chat functionality and post detail page"
                                    className="securityMockup"
                                    src="security-mockups.png"
                                  />
                                </div>
                                <div
                                  className="right"
                                >
                                  <h2
                                    className="heading"
                                  >
                                    Your safety and security are important to us
                                  </h2>
                                  <p
                                    className="subheading"
                                  >
                                    We will never share any of your personal information. The other users in your neighborhood will only see your first name and location radius (not your exact location), so the amount of information that you share in the chat is 100% your call.
                                  </p>
                                </div>
                              </section>
                              <section
                                className="section last"
                              >
                                <h2
                                  className="heading"
                                >
                                  Ready to make a difference in your community?
                                </h2>
                                <div
                                  className="buttonSection"
                                >
                                  <button
                                    className="signupButton"
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    Start now
                                  </button>
                                  <button
                                    className="demoButton"
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    See a demo
                                  </button>
                                </div>
                              </section>
                              <footer
                                className="footer"
                              >
                                <div
                                  className="footerDetails"
                                >
                                  <p>
                                    Made by Zoe Ferencova
                                  </p>
                                  <a
                                    href="https://github.com/zoeferencova/"
                                    rel="noopener noreferrer"
                                    target="_blank"
                                  >
                                    <i
                                      className="fab fa-github"
                                    />
                                  </a>
                                  <a
                                    href="https://www.linkedin.com/in/zoeferencova/"
                                    rel="noopener noreferrer"
                                    target="_blank"
                                  >
                                    <i
                                      className="fab fa-linkedin"
                                    />
                                  </a>
                                </div>
                                <img
                                  alt="small logo"
                                  className="securityMockup"
                                  src="light-logo.png"
                                />
                              </footer>
                            </main>
                          </LandingPage>
                        </withRouter(LandingPage)>
                      </component>
                    </Route>
                  </Switch>
                </ErrorBoundary>
              </withRouter(ErrorBoundary)>
            </main>
          </UnauthenticatedApp>
        </main>
      </App>
    </withRouter(App)>
  </Router>
</BrowserRouter>
`;
